src/Cli/dotnet/CommonOptions.cs

```diff
@@ -100,6 +116,57 @@ public static bool VerbosityIsDetailedOrDiagnostic(this VerbosityOptions verbosi
                 verbosity.Equals(VerbosityOptions.d) ||
                 verbosity.Equals(VerbosityOptions.detailed);
         }
+
+        internal static IEnumerable<string> ResolveArchOptionToRuntimeIdentifier(string arg, ParseResult parseResult)
+        {
+            if (parseResult.BothArchAndOsOptionsSpecified())
+            {
+                return Array.Empty<string>();
+            }
+
+            return ResolveRidShorthandOptions(null, arg);
+        }
+
+        internal static IEnumerable<string> ResolveOsOptionToRuntimeIdentifier(string arg, ParseResult parseResult)
+        {
+            if (parseResult.BothArchAndOsOptionsSpecified())
+            {
+                return ResolveRidShorthandOptions(arg, parseResult.ValueForOption<string>(CommonOptions.ArchitectureOption().Aliases.First()));
+            }
+
+            return ResolveRidShorthandOptions(arg, null);
+        }
+
+        private static IEnumerable<string> ResolveRidShorthandOptions(string os, string arch)
+        {
+            var currentRid = GetCurrentRuntimeId();
+            os = string.IsNullOrEmpty(os) ? GetOsFromRid(currentRid) : os;
+            arch = string.IsNullOrEmpty(arch) ? GetArchFromRid(currentRid) : arch;
+            var properties = new string[] { $"-property:RuntimeIdentifier={os}-{arch}" };
+            return properties;
+        }
+
+        private static string GetCurrentRuntimeId()
+        {
+            var dotnetRootPath = Path.GetDirectoryName(Environment.ProcessPath);
+            dotnetRootPath = dotnetRootPath.Contains("dotnet") ? dotnetRootPath : Path.Combine(dotnetRootPath, "dotnet");
+            var ridFileName = "NETCoreSdkRuntimeIdentifierChain.txt";
+            string runtimeIdentifierChainPath = string.IsNullOrEmpty(Product.Version) ?
+                Path.Combine(Directory.GetDirectories(Path.Combine(dotnetRootPath, "sdk"))[0], ridFileName) :
+                Path.Combine(dotnetRootPath, "sdk", Product.Version, ridFileName);
+            string[] currentRuntimeIdentifiers = File.Exists(runtimeIdentifierChainPath) ?
+                File.ReadAllLines(runtimeIdentifierChainPath).Where(l => !string.IsNullOrEmpty(l)).ToArray() :
+                new string[] { };
+            if (currentRuntimeIdentifiers == null || !currentRuntimeIdentifiers.Any() || !currentRuntimeIdentifiers[0].Contains("-"))
+            {
+                throw new GracefulException(CommonLocalizableStrings.CannotResolveRuntimeIdentifier);
+            }
+            return currentRuntimeIdentifiers[0]; // First rid is the most specific (ex win-x64)
```

Comment:
That logic is to account for the slightly different configuration when we're testing, this is how we're able to pick up the RID file in tests.

Reply:
Does it mean that we're not testing how the product will work when installed for real? That would be unfortunate... and would be really good to fix.
What is the layout on disk the tests use? And why it can't be changed to look like if the product is correctly installed.